src/FnTools/Types/Result.cs

```diff
@@ -0,0 +1,261 @@
+using System;
+using System.Collections.Generic;
+using FnTools.Exceptions;
+using FnTools.Types.Interfaces;
+using static FnTools.Prelude;
+
+namespace FnTools.Types
+{
+    public readonly struct Result<TOk, TError> : IGettable<TOk>, IEquatable<Result<TOk, TError>>, IToOption<TOk>
+    {
+        private readonly TError _error;
+
+        private readonly TOk _ok;
+
+        public bool IsOk { get; }
+
+        public bool IsError => !IsOk;
+        
+        public Option<TError> Error => IsError ? Some(_error) : None;
+
+        public Option<TOk> Ok => IsOk ? Some(_ok) : None;
```

Comment:
it's conveniently when you have both "Error" and "Ok" (ToOption), than only "Error" and "ToOption"

Reply:
I think it is pointless because you have pack of GetXXX methods to receive TOk value, all MapXXXs a done over TOk and even more if you want Option<TOk> you do it with ToOption()